rs/config/src/embedders.rs

```diff
@@ -100,6 +100,33 @@ const STABLE_MEMORY_ACCESSED_PAGE_LIMIT_QUERY: NumOsPages =
 /// also used as the maximum size for the Wasm chunk store of each canister.
 pub const WASM_MAX_SIZE: NumBytes = NumBytes::new(100 * 1024 * 1024); // 100 MiB
 
+/// Best-effort responses feature rollout stage.
+///
+/// The intent is to incrementally release the feature, first to application
+/// subnets only, then to all subnets, by rolling forward one stage at a time.
+/// Rolling back is also supported, including directly to stage 1. Rolling back
+/// to stage 0 will break all canisters that have started using best-effort
+/// response calls, as they will trap reliably.
+//
+// TODO(MR-649): Drop this once best-effort responses are fully rolled out.
+#[derive(Copy, Clone, Eq, PartialEq, Debug, Deserialize, Serialize)]
+pub enum BestEffortResponsesFeature {
+    /// Stage 0: Feature is disabled, API calls trap.
+    DisabledByTrap,
+
+    /// Stage 1: Feature is disabled, `ic0_call_with_best_effort_response` API is a
+    /// no-op, silently falling back to a guaranteed response.
+    FallBackToGuaranteedResponse,
+
```

Comment:
It would be nice to have an additional step where we could enable it for a selected list of subnets only as a first rollout step, maybe something like
`ListedSubnetsOnly(Vec<SubnetId>)` or something?

Reply:
Done.